### PR TITLE
Aging Validation

### PR DESCRIPTION
**Pull request by pressel**
This merge request adds improved cross-validation for aging. In particular validation tests now fail if tolerances are exceeded. 

That said, due to what appears to be the poor conditioning of the aging parameterization, large tolerances are required for aerosol number variables. These tolerances are set in compare_aging_validation.py. 

This merge request also fixes some differences in constants between E3SM and mam4xx.

### CODE REVIEW

**Comment by jeff-cohere** [comment]
It looks like we've got a failing test in a double precision build:

```
...
test 64
      Start 64: validate_pcarbon_aging_1subarea

64: Test command: /opt/hostedtoolcache/Python/3.11.2/x64/bin/python3 "compare_aging_validation.py" "mam4xx_pcarbon_aging_1subarea.py" "mam_pcarbon_aging_1subarea.py"
64: Working Directory: /home/runner/work/mam4xx/mam4xx/build/src/validation/aging
64: Test timeout computed to be: 1500
64: qaer_cur: L1 = 3.44967e-14, L2 = 5.02662e-14, Linf = 9.24203e-14
64: qaer_del_coag: L1 = 4.859e+254, L2 = 4.859e+254, Linf = 4.859e+254
64: Traceback (most recent call last):
64:   File "/home/runner/work/mam4xx/mam4xx/build/src/validation/aging/compare_aging_validation.py", line 82, in <module>
64:     assert np.allclose(o1, o2)
64: AssertionError
64/72 Test #64: validate_pcarbon_aging_1subarea .......................***Failed    0.11 sec
qaer_cur: L1 = 3.44967e-14, L2 = 5.02662e-14, Linf = 9.24203e-14
qaer_del_coag: L1 = 4.859e+254, L2 = 4.859e+254, Linf = 4.859e+254
Traceback (most recent call last):
  File "/home/runner/work/mam4xx/mam4xx/build/src/validation/aging/compare_aging_validation.py", line 82, in <module>
    assert np.allclose(o1, o2)
AssertionError
...
```

File: src/validation/aging/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 set(AGING_VALIDATION_DIR ${MAM_X_VALIDATION_DIR}/aging)
-set(AGING_VALIDATION_SCRIPTS_DIR ${MAM_X_VALIDATION_DIR}/scripts)
+set(AGING_VALIDATION_SCRIPTS_DIR ${MAM_X_VALIDATION_DIR}/aging)
```
Comment (jeff-cohere) [inline src/validation/aging/CMakeLists.txt:2]: Do you think this is still needed (as opposed to `AGING_VALIDATION_DIR`)?

Comment (pressel) [reply src/validation/aging/CMakeLists.txt:2]: You are correct that is not needed. Removing.

File: src/mam4xx/aging.hpp [hunk 2]
```diff
@@ -74,73 +73,81 @@ void mam_pcarbon_aging_frac(
...
   Real vol_core = 0.0;
   for (int mi = 0; mi < 3; ++mi) {
     const int ispec = spec_modes[mi];
-    vol_core += qaer_cur[ispec][nsrc] * core_volumes[mi];
+    vol_core += qaer_cur[ispec][imom_pc] * core_volumes[mi];
   }
 
   const Real fac_volsfc = haero::exp(
-      2.5 * haero::log(haero::pow(mam4::modes(nsrc).mean_std_dev, 2.0)));
+      2.5 * haero::pow(haero::log(mam4::modes(imom_pc).mean_std_dev), 2.0));
```
Comment (jeff-cohere) [inline src/mam4xx/aging.hpp:145]: you could replace this `pow` expression with `haero::square(haero::log(mam4::modes(imom_pc).mean_std_dev))` if you wanted.

**Review by jeff-cohere — APPROVED** [review]
Looks good!

File: src/mam4xx/aging.hpp (hunk 2 above)
Comment (pressel) [reply src/mam4xx/aging.hpp]: Done.

**Pull request merged**